src/aktualizr_secondary/main.cc

```diff
@@ -88,7 +88,14 @@ int main(int argc, char *argv[]) {
     LOG_DEBUG << "Current directory: " << boost::filesystem::current_path().string();
 
     auto secondary = AktualizrSecondaryFactory::create(config);
-    SecondaryTcpServer(*secondary, config.network.primary_ip, config.network.primary_port, config.network.port).run();
+    SecondaryTcpServer tcp_server(*secondary, config.network.primary_ip, config.network.primary_port,
+                                  config.network.port, config.uptane.force_install_completion);
+
+    tcp_server.run();
+
+    if (tcp_server.exit_reason() == SecondaryTcpServer::ExitReason::kRebootNeeded) {
+      secondary->completeInstall();
```

Comment:
How can we guarantee here that the TCP/IP stack has sent all data from its internal buffers before the reboot is called?
It looks like there is missing connection shutdown in HandleOneConnection() and connection socket closing in run().

Reply:
I've added a RAII socket in `run()`.